### PR TITLE
chore(fleet): Add "direct" RC option to pull configurations

### PR DESCRIPTION
**Pull request by BaptisteFoy**
### What does this PR do?
Real CDN is still WIP so we need another reliable option to pull config: this PR adds a regular RC client to do so. It also adds an option to use the real CDN that is disabled by default

### Additional Notes
Config field to allow CDN use in the daemon will come in another PR

### CODE REVIEW

**Review by coignetp — APPROVED** [review]

File: pkg/fleet/daemon/daemon.go [hunk 1]
```diff
@@ -41,7 +41,7 @@ const (
 	// gcInterval is the interval at which the GC will run
 	gcInterval = 1 * time.Hour
 	// refreshStateInterval is the interval at which the state will be refreshed
-	refreshStateInterval = 1 * time.Minute
+	refreshStateInterval = 30 * time.Second
```
[the remark this answers is not in the document]
Comment (BaptisteFoy) [reply pkg/fleet/daemon/daemon.go:44]: It's mostly to make the feedback loop faster, now that we actually need to report the config state change.

File: pkg/fleet/daemon/daemon.go [hunk 2]
```diff
@@ -602,9 +602,6 @@ func (d *daemonImpl) refreshState(ctx context.Context) {
 		}
 
 		configVersion, err := d.resolveRemoteConfigVersion(ctx, pkg)
-		if err != nil {
-			log.Errorf("could not get agent remote config version: %v", err)
-		}
 		if err == nil {
 			p.RemoteConfigVersion = configVersion
 		} else if err != cdn.ErrProductNotSupported {
```
[the remark this answers is not in the document]
Comment (BaptisteFoy) [reply pkg/fleet/daemon/daemon.go:607]: We don't want to log for `ErrProductNotSupported`, it's expected

File: pkg/fleet/internal/cdn/cdn.go [hunk 3]
```diff
@@ -40,8 +41,17 @@ type CDN interface {
 
 // New creates a new CDN.
 func New(env *env.Env, configDBPath string) (CDN, error) {
+	if runtime.GOOS == "windows" {
```
[the remark this answers is not in the document]
Comment (BaptisteFoy) [reply pkg/fleet/internal/cdn/cdn.go:45]: Done in [more comments](https://github.com/DataDog/datadog-agent/pull/30369/commits/35b277b62903f27194bbf3f40b8f903d3dbbc09d)!

File: pkg/fleet/internal/cdn/cdn_regular.go [hunk 4]
```diff
@@ -20,12 +20,12 @@ import (
 	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
 )
 
-type cdnRemote struct {
+type cdnRegular struct {
```
Comment (mellon85) [inline pkg/fleet/internal/cdn/cdn_regular.go:23]: Can't you call them 
- CDNNoOp
- CDN
- CDNRCClient

File: pkg/fleet/internal/cdn/cdn_noop.go [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+	return &cdnNoop{}, nil
+}
+
+// Get gets the configuration from the CDN.
+func (c *cdnNoop) Get(_ context.Context, _ string) (Config, error) {
+	return &configNoop{}, nil
+}
+
+func (c *cdnNoop) Close() error {
+	return nil
```
[the remark this answers is not in the document]
Comment (BaptisteFoy) [reply pkg/fleet/internal/cdn/cdn_noop.go:32]: Sure, done in [65055a5](https://github.com/DataDog/datadog-agent/pull/30369/commits/65055a58be1c564020787e82bc140bcf2c8df94d)

File: pkg/fleet/internal/cdn/cdn_regular.go (hunk 4 above)
Comment (BaptisteFoy) [reply pkg/fleet/internal/cdn/cdn_regular.go]: I don't have a strong opinion. Done in [65055a5](https://github.com/DataDog/datadog-agent/pull/30369/commits/65055a58be1c564020787e82bc140bcf2c8df94d)

**Review by mellon85 — APPROVED** [review]

**Pull request merged**